src/com/dotcms/api/system/user/UserServiceFactory.java

```diff
@@ -0,0 +1,366 @@
+				throws Exception {
+			int start = 0;
+			if (params.containsKey("start")) {
+				start = Integer.parseInt((String) params.get("start"));
+			}
+			int limit = -1;
+			if (params.containsKey("limit")) {
+				limit = Integer.parseInt((String) params.get("limit"));
+			}
+			String query = "";
+			if (params.containsKey("query")) {
+				query = (String) params.get("query");
+			}
+			boolean includeAnonymous = false;
+			if (params.containsKey("includeAnonymous")) {
+				includeAnonymous = Boolean.valueOf((String) params.get("includeAnonymous"));
+			}
+			// Initially set to "true" for backwards compatibility
+			boolean includeDefault = true;
+			if (params.containsKey("includeDefault")) {
+				includeDefault = Boolean.valueOf((String) params.get("includeDefault"));
+			}
+			Map<String, Object> results;
+			if ((InodeUtils.isSet(assetInode) && !assetInode.equals("0"))
+					&& (UtilMethods.isSet(permission) && !permission.equals("0"))) {
+				results = processUserListWithPermissionOnInode(assetInode, permission, query, start, limit);
+			} else {
+				results = processUserList(query, start, limit, includeAnonymous, includeDefault);
+			}
+			return results;
+		}
+
+		/**
+		 * Returns a {@link Map} containing a list of dotCMS {@link User}
+		 * objects based on the specified search criteria.
+		 * 
+		 * @param query
+		 *            - The string or characters that are part of the first
+		 *            name, last name, or e-mail of the user.
+		 * @param start
+		 *            - For pagination purposes. Lower range of the set to
+		 *            include in the query result.
+		 * @param limit
+		 *            - For pagination purposes. Upper range of the set to
+		 *            include in the query result.
+		 * @param includeAnonymous
+		 *            - If set to {@code true}, the "Anonymous" user will be
+		 *            included in the result. Otherwise, set to {@code false}.
+		 * @param includeDefault
+		 *            - If set to {@code true}, the "Default" user will be
+		 *            included in the result. Otherwise, set to {@code false}.
+		 * @return A Map containing the user list and additional query
+		 *         information.
+		 */
+		private Map<String, Object> processUserList(String query, int start, int limit, boolean includeAnonymous,
+				boolean includeDefault) {
+			Map<String, Object> results = new UsersListTemplate("", 0, query, start, limit, includeAnonymous, includeDefault) {
+				UserAPI userAPI = APILocator.getUserAPI();
+
+				@Override
+				public int getUserCount() {
+					try {
+						return new Long(userAPI.getCountUsersByNameOrEmailOrUserID(this.filter, this.includeAnonymous,
+								this.includeDefault)).intValue();
+					} catch (DotDataException e) {
+						Logger.error(this, e.getMessage(), e);
+						return 0;
+					}
+				}
+
+				@Override
+				public List<User> getUsers() {
+					try {
+						int page = (start / limit) + 1;
```

Comment:
take care with limit as zero, should be validate (if it is not already validated that can not be zero)